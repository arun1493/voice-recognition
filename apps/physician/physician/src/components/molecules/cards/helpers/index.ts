export * from "./labels";
export * from "./formatters";
