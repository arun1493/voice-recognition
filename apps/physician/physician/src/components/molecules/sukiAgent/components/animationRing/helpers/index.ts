export * from "./ringSize";
export * from "./animation";
