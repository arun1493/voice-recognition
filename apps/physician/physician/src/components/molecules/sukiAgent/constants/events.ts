import { ClientReceiveEventTypes } from "@suki-web/suki-business";

import {
  handleCompositionCreationInNotepad,
  handleCreateCompositionAcknowledgement,
  handleIntent,
  handleResourceExhaustedError,
  handleTurnOffAgent,
  handleTurnOnAgent
} from "../eventHandlers";

const AGENT_TURN_ON_EVENT = {
  type: ClientReceiveEventTypes.TURN_ON_AGENT,
  handler: handleTurnOnAgent
};

const AGENT_TURN_OFF_EVENT = {
  type: ClientReceiveEventTypes.TURN_OFF_AGENT,
  handler: handleTurnOffAgent
};

const INTENT_EVENT = {
  type: ClientReceiveEventTypes.INTENT,
  handler: handleIntent
};

const COMPOSITION_CREATED_IN_NOTEPAD_EVENT = {
  type: ClientReceiveEventTypes.COMPOSITION_CREATED_IN_NOTEPAD,
  handler: handleCompositionCreationInNotepad
};

const CREATE_COMPOSITION_ACKNOWLEDGEMENT_EVENT = {
  type: ClientReceiveEventTypes.CREATE_COMPOSITION_ACKNOWLEDGEMENT,
  handler: handleCreateCompositionAcknowledgement
};

const RESOURCE_EXHAUSTED_ERROR_EVENT = {
  type: ClientReceiveEventTypes.RESOURCE_EXHAUSTED_ERROR,
  handler: handleResourceExhaustedError
};

const EVENTS = [
  AGENT_TURN_OFF_EVENT,
  AGENT_TURN_ON_EVENT,
  COMPOSITION_CREATED_IN_NOTEPAD_EVENT,
  CREATE_COMPOSITION_ACKNOWLEDGEMENT_EVENT,
  INTENT_EVENT,
  RESOURCE_EXHAUSTED_ERROR_EVENT
];

export { EVENTS };
