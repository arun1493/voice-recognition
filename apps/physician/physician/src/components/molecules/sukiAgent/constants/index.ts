export * from "./sounds";
export * from "./events";
export * from "./general";
