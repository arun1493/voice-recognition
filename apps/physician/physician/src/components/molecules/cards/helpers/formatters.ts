import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
