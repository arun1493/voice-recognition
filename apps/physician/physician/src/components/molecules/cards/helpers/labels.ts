import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
