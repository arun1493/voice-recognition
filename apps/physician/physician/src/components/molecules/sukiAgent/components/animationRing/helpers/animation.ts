import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
