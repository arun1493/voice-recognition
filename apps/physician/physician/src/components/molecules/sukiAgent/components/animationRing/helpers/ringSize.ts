import { SukiAgentRingWidth, SukiAgentSize } from "../../../../../../constants";

function getRingWidth(agentSize: SukiAgentSize, agentPosition: "center" | "bottom") {
  if (agentPosition === "bottom") return SukiAgentRingWidth[SukiAgentSize.SM];
  return SukiAgentRingWidth[agentSize] || SukiAgentRingWidth[SukiAgentSize.XL];
}

export { getRingWidth };
