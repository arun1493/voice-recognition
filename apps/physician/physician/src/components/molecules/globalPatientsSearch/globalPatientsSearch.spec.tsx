import "@testing-library/jest-dom/extend-expect";
import { cleanup, fireEvent, RenderResult } from "@testing-library/react";

import { Patient } from "../../../types";
import { componentSetup } from "../../../hocs";
import { configureI18n } from "../../../helpers";
import { mockPatients } from "../../../constants";
import { GlobalPatientsSearch, GlobalPatientsSearchProps } from ".";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const handleSearch = (t: string) => null;
const mockClick = jest.fn((patient: Patient) => console.log("handlePatientClick"));

const props: GlobalPatientsSearchProps = {
  hasCloseInInput: true,
  onPatientClick: mockClick,
  onSearch: handleSearch,
  patients: mockPatients,
  prefixIcon: "search",
  value: ""
};
const setup = componentSetup(GlobalPatientsSearch, props);
describe("GlobalPatientsSearch", () => {
  let component: RenderResult;

  beforeEach(() => {
    configureI18n();
    component = setup();
  });

  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(GlobalPatientsSearch, props);
  });

  it("should respond to input change", () => {
    const { baseElement } = component;
    const input = baseElement.querySelector("[data-testid=input]") as HTMLInputElement;

    expect(input.value).toBe("");

    fireEvent.change(input, { target: { value: "test" } });

    expect(input.value).toBe("test");
  });

  it("should render results", () => {
    const { baseElement } = component;
    const input = baseElement.querySelector("[data-testid=input]") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "test" } });
    input.focus();
    const listItems = baseElement.querySelectorAll("[data-testid=listItem]");
    expect(listItems.length).toBe(3);
  });

  //TODO: Create an e2e test for this. It currently fails. I think click events only work when applied directly to the styled component (html element).
  // it("should handle click on search result item", async () => {
  //   const { baseElement } = component;

  //   const input = baseElement.querySelector("[data-testid=input]") as HTMLInputElement;

  //   fireEvent.change(input, { target: { value: "test" } });

  //   input.focus();
  //   const listItems = screen.getAllByTestId("listItem");
  //   expect(listItems.length).toBe(3);
  //   const listItem = listItems[0] as HTMLElement;
  //   fireEvent.click(listItem);
  //   // listItem.click();

  //   expect(mockClick).toBeCalled();
  // });
});
