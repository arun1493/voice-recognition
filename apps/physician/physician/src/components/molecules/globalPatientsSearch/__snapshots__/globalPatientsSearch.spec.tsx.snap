// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalPatientsSearch should match snapshot 1`] = `
<div
  className="globalPatientsSearchstyled__StyledPatientsSearch-nftnqf-0 eIhoVw"
>
  <div
    className="globalPatientsSearchstyled__StyledSearchContainer-nftnqf-1 BuUsd"
  >
    <div
      className="globalPatientsSearchstyled__StyledSearchIcon-nftnqf-3 tfHfz"
    >
      <i
        className="SukiIcon__StyledIcon-q9b9ar-0 hViFDz sukiIcon "
        color="secondary"
        size="md"
      >
        <svg
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.842 3.043a6.191 6.191 0 0 0-2.022.548c-1.972.898-3.326 2.651-3.73 4.829-.092.495-.092 1.665 0 2.16.514 2.771 2.559 4.816 5.33 5.33.495.092 1.665.092 2.16 0 2.771-.514 4.816-2.559 5.33-5.33.092-.495.092-1.665 0-2.16-.451-2.433-2.09-4.326-4.41-5.095a6.485 6.485 0 0 0-2.658-.282m1.218 2.001a4.52 4.52 0 0 1 3.48 2.493c.721 1.489.559 3.304-.411 4.606-.582.782-1.313 1.319-2.2 1.616a4.4 4.4 0 0 1-2.858 0c-1.981-.664-3.253-2.632-3.038-4.699a4.484 4.484 0 0 1 5.027-4.016m3.86 10.471c-.3.098-.554.326-.671.602-.076.181-.075.542.002.729.065.154 2.931 4.263 3.099 4.442.252.269.742.359 1.096.202.217-.096.423-.298.519-.509.09-.197.097-.57.016-.765-.069-.165-2.893-4.212-3.077-4.41a1.26 1.26 0 0 0-.281-.214c-.172-.088-.544-.129-.703-.077"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </i>
    </div>
    <div
      className="formInputstyled__StyledInputContainer-p3pq9x-0 jOohTQ"
    >
      <input
        className="formInputstyled__StyledFormInput-p3pq9x-1 ckqHZa globalPatientsSearchstyled__StyledSearchInput-nftnqf-2 daWdeC"
        data-cy="form-input"
        data-testid="input"
        disabled={false}
        onChange={[Function]}
        placeholder="Enter Patient Name"
        type="text"
      />
    </div>
  </div>
  <ul
    className="globalPatientsSearchstyled__StyledPatientsListDropDown-nftnqf-4 bfwyNs"
    data-cy="patients-list-dropdown"
  >
    <li
      className="sukiListItemstyled__StyledSukiListItem-lvcifx-0 duNFrf"
      data-testid="listItem"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sukiListItemstyled__StyledSukiTitleAreaWrapper-lvcifx-1 lokoIV"
        onClick={[Function]}
      >
        <div
          className="sukiListItemTitlestyled__StyledSukiListItemTitle-sc-1om802d-0 kinFhQ"
        >
          TestB TestC
        </div>
        <div
          className="sukiListItemSubtitlestyled__StyledSukiListItemSubtitle-sc-1txtyeo-0 fREnXW"
        >
          5 months • F • MRN ,865
        </div>
      </div>
    </li>
    <li
      className="sukiListItemstyled__StyledSukiListItem-lvcifx-0 duNFrf"
      data-testid="listItem"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sukiListItemstyled__StyledSukiTitleAreaWrapper-lvcifx-1 lokoIV"
        onClick={[Function]}
      >
        <div
          className="sukiListItemTitlestyled__StyledSukiListItemTitle-sc-1om802d-0 kinFhQ"
        >
          Test Patient
        </div>
        <div
          className="sukiListItemSubtitlestyled__StyledSukiListItemSubtitle-sc-1txtyeo-0 fREnXW"
        >
          5 months • M • MRN ,12345
        </div>
      </div>
    </li>
    <li
      className="sukiListItemstyled__StyledSukiListItem-lvcifx-0 duNFrf"
      data-testid="listItem"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sukiListItemstyled__StyledSukiTitleAreaWrapper-lvcifx-1 lokoIV"
        onClick={[Function]}
      >
        <div
          className="sukiListItemTitlestyled__StyledSukiListItemTitle-sc-1om802d-0 kinFhQ"
        >
          Test 10
        </div>
        <div
          className="sukiListItemSubtitlestyled__StyledSukiListItemSubtitle-sc-1txtyeo-0 fREnXW"
        >
          -- • F • MRN ,12345
        </div>
      </div>
    </li>
  </ul>
</div>
`;
